src/SFA.DAS.PR.Application/EmployerRelationships/Queries/GetProviderEmployerRelationship/GetProviderEmployerRelationshipQueryHandler.cs

```diff
@@ -0,0 +1,45 @@
+﻿using MediatR;
+using SFA.DAS.PR.Application.Mediatr.Responses;
+using SFA.DAS.PR.Domain.Entities;
+using SFA.DAS.PR.Domain.Extensions;
+using SFA.DAS.PR.Domain.Interfaces;
+
+namespace SFA.DAS.PR.Application.EmployerRelationships.Queries.GetProviderEmployerRelationship;
+
+public class GetProviderEmployerRelationshipQueryHandler(IAccountProviderLegalEntitiesReadRepository accountProviderLegalEntitiesReadRepository, IPermissionAuditReadRepository permissionAuditReadRepository, IRequestReadRepository requestReadRepository) : IRequestHandler<GetProviderEmployerRelationshipQuery, ValidatedResponse<GetProviderEmployerRelationshipQueryResult?>>
+{
+    public async Task<ValidatedResponse<GetProviderEmployerRelationshipQueryResult?>> Handle(GetProviderEmployerRelationshipQuery query, CancellationToken cancellationToken)
+    {
+        AccountProviderLegalEntity? accountProviderLegalEntity = await accountProviderLegalEntitiesReadRepository.GetAccountProviderLegalEntityByProvider(query.Ukprn!.Value, query.AccountLegalEntityId!.Value, cancellationToken);
+
+        if(accountProviderLegalEntity == null)
+        {
+            return ValidatedResponse<GetProviderEmployerRelationshipQueryResult?>.EmptySuccessResponse();
+        }
+
+        GetProviderEmployerRelationshipQueryResult result = (GetProviderEmployerRelationshipQueryResult)accountProviderLegalEntity;
+
+        PermissionsAudit? permissionAudit = await permissionAuditReadRepository.GetMostRecentPermissionAudit(query.Ukprn!.Value, query.AccountLegalEntityId!.Value, cancellationToken);
+
+        if(permissionAudit != null)
+        {
+            result.LastAction = EnumExtensions.ToEnum<PermissionAction>(permissionAudit.Action);
+            result.LastActionTime = permissionAudit.Eventtime;
+        }
+        else
+        {
+            result.LastActionTime = accountProviderLegalEntity.Updated ?? accountProviderLegalEntity.Created;
```

Comment:
If there was no last action, why are we using legal entity dates, it can be of no relevance.

Reply:
@DevenKShah It is what is suggested in the documentation.

**If there is no historical PermissionsAudit  records for the relationship, the "lastActionTime" date will be taken from AccountProviderLegalEntities Created or Updated date.**

Reply:
But why? The LastAction is being left null and only the date is being produced. Does not make sense. @alandburns can you fill in please?